### PR TITLE
npm integration 

### PR DESCRIPTION
**Pull request by yaitskov**
Hi,

I am not a java script professional developer but even for me it looks ugly.
A framework generating java script code should talk to npm.
It is not convenient to extract java script files out of jars.

I check npmjs.com and there is no jsweet candies.

So I created a project java-jsweet there and [uploaded](https://www.npmjs.com/package/j4ts).
As me for permission if you need to publish update on your own.

In short npm for web developer is a super tool like maven for Java programmer.
Dependencies in JavaScript world are provided through npm.

I hope this contribution would help in gaining jsweet more popularity.

### CODE REVIEW

**Comment by yaitskov** [comment]
Oh. I don't how do you use /dist, but I strongly advise to consider removing generated code out of version control and locating all generated files under /target which compliant with maven policy.

With npm repository there is no need to keep javascript in repo and in maven artifacts either.

**Comment by renaudpawlak** [comment]
Many thanks for that excellent initiative/contribution. That's great.
Yeah, I guess that the dist folder should be removed.
I am just surprise you don't need to compile J4TS using comings modules to publish it to npm... but my knowledge of npm is quite limited.
